examples/teleport-usage/teleport_usage.py

```diff
@@ -0,0 +1,110 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import json
+import datetime
+import boto3
+import os
+
+from boto3.dynamodb.conditions import Key, Attr
+
+
+# Check for AWS DynamoDB table name and region ENV, if none prompt
+table_name = os.getenv("TABLE_NAME")
+if not table_name:
+    table_name = input("DynamoDB events table name: ")
+
+aws_region = os.getenv("AWS_REGION")
+if not aws_region:
+    aws_region = input("AWS Region for DynamoDB: ")
+
+# Check ENV for start time, end time is always 30 days later.
+start_date = os.getenv("START_DATE")
+if not start_date:
+    now = datetime.datetime.now()
+    start_date = now - datetime.timedelta(days=30)
+else:
+    start_date = datetime.datetime.strptime(start_date,'%Y-%m-%d')
+
+end_date = start_date + datetime.timedelta(days=30)
+
+# Configure DynamoDB for query
+dynamodb = boto3.resource("dynamodb", region_name=aws_region)
+table = dynamodb.Table(table_name)
+
+# Generating data message
+start_date_str = start_date.strftime("%b %d %y")
+end_date_str = end_date.strftime("%b %d %y")
+day_range = (end_date - start_date).days
+d_range = (end_date - start_date).days + 1
+print("""
+NOTE: This tool queries usage for the last 30 days")
+For a custom start date use ENV START_DATE (format YYYY-MM-DD)")
+""")
+print(f"Gathering data for unique users who accessed each protocol between {start_date_str} to {end_date_str} ({day_range} days): ")
+
+# Creating a set per protocol to gather unique users
+unique_users = {
+    'app.session.start': set(),
+    'db.session.start': set(),
+    'kube.request': set(),
+    'session.start': set(),
+    'windows.desktop.session.start': set(),
+    
+}
+
+# Run Dynamodb query to retrieve all login for SSH for specified date range
+for i in range(d_range):
+    current_date = start_date + datetime.timedelta(days=i)
+
+    date = current_date.strftime("%Y-%m-%d")
+    response = table.query(
+        TableName=table_name,
+        IndexName="timesearchV2",
+        KeyConditionExpression=Key("CreatedAtDate").eq(date),
+        FilterExpression=Attr("EventType").contains('session.start' or 'kube.request'),
+    )
+    # Parsing the response from DynamoDB. Separating users by event type
+    for item in response['Items']:
+        event_type = item['FieldsMap']['event']
+        user = item['FieldsMap']['user']
+        if event_type in unique_users:
+            # For SSH access, ensuring there is not a double count the Kube execs
+            if event_type == 'session.start' and item['FieldsMap'].get('kubernetes_cluster', ''):
+                continue  # skip if kubernetes_cluster exists and is not empty

```

Comment:
I have a question about a [previous comment](https://github.com/gravitational/teleport/pull/20621#discussion_r1104939088) related to this.

> In Teleport 10, yes. In Teleport 11, first triggers a GET kube.request on the pod so yes it is captured, but then the exec sessions actually show up just like an SSH session start which may lead to an overcount on SSH as they use the same event type.

I checked Teleport 9, 10, 11, and 12 and there is always appears to be a `kube.request` associated with a `kubectl exec`. Was there a time when this was not the case?

Reply:
Ping about this question.